lib/perl/Genome/Model/Tools/Bsmap/MethCalcConversionRate.pm

```diff
@@ -0,0 +1,263 @@
+# % Last Change: Thu Jan 15 01:21:18 PM 2015 CST
+package Genome::Model::Tools::Bsmap::MethCalcConversionRate;
+
+use strict;
+use warnings;
+use Genome;
+
+class Genome::Model::Tools::Bsmap::MethCalcConversionRate {
+    is => 'Command',
+    has => [
+        snvs_file => {
+            is => 'String',
+            is_optional => 1,
+            doc => 'Use snvs.hq file to calculate methylation conversion',
+        },
+        model_id => {
+            is => 'String',
+            is_optional => 1,
+            doc => 'Use genome model ID to calculate methylation conversion',
+        },
+        output_file => {
+            is => 'String',
+            is_optional => 1,
+            doc => 'Output methylation conversion',
+        },
+    ],
+};
+
+sub help_synopsis {
+  return <<EOS
+    gmt bsmap meth-calc-conversion-rate --model-id=394d6228a7b5487a9cb0ad0c448b5a44
+
+    gmt bsmap meth-calc-conversion-rate --snvs-file=/gscmnt/gc9016/info/model_data/394d6228a7b5487a9cb0ad0c448b5a44/buildf92b6057072948c2a6056a3ee412d596/variants/snv/meth-ratio-2.74-d41d8cd98f00b204e9800998ecf8427e/MT/snvs.hq
+
+EOS
+}
+
+sub help_brief {
+    "calculate methylation conversion using Mitochondrial DNA or spiked lambda"
+}
+
+sub help_detail {
+    "calculate methylation conversion using Mitochondrial DNA or spiked lambda"
+}
+
+sub bs_rate {
+    my $count = 0;
+    my $totalreads = 0;
+    my $methreads = 0;
+
+    open (my $fh, '<', $_[0]) or die("Could not open snvs file.");
+    while (my $line = <$fh>) {
+        if ( $count == 0 ) {
+            $count = 1;
+            next;
+        }
+        chomp($line);
+        my @F = split("\t",$line);
```

Comment:
Parsing of files should generally not be done ad-hoc like this.  We have parser library for dealing with tab-delimited data.  It may also be beneficial to abstract parsing of this type of file so that it can be re-used.